pubsub/gcp/gcp.go

```diff
@@ -133,15 +130,26 @@ var _ pubsub.Publisher = &publisher{}
 var _ pubsub.MultiPublisher = &publisher{}
 
 // NewPublisher will instantiate a new GCP MultiPublisher.
-func NewPublisher(ctx context.Context, projID, topic string, opts ...option.ClientOption) (pubsub.MultiPublisher, error) {
-	client, err := gpubsub.NewClient(ctx, projID, opts...)
+func NewPublisher(ctx context.Context, cfg Config) (pubsub.MultiPublisher, error) {
+	p := &publisher{}
+
+	if cfg.ProjectID == "" {
+		return p, errors.New("project id is required")
```

Comment:
https://github.com/NYTimes/gizmo/pull/147/files/c205e43b0c5df6d6c5a9597bee3c1389baa5264c#r193798878

@jprobinson How do you make this then? The same is being made on pubsub/aws/aws.go:NewPublisher